boot/zephyr/include/flash_map/flash_map.h

```diff
@@ -95,6 +95,14 @@ struct flash_sector {
 };
 
 /*
+ * Initialize the flash partition map description array according to
+ * the image index. Three memory partition belongs to each bootable
+ * image: sw_slot_0(primary slot); sw_slot_1(secondary_slot);
+ * scratch_area.
+ */
+void flash_partition_map_init(uint8_t img_idx);
```

Comment:
This will break the build for mynewt. The bootutil code is the core code shared across the operating systems. This change only affects Zephyr's copy of flash_map.h. Thus, this API change would need to be discussed with the other supported RTOSes.

Is it possible to implement this without adding this function at all? I believe so; see comments elsewhere.